src/database/DataTypes/ItemBatch.js

```diff
@@ -145,6 +150,20 @@ export class ItemBatch extends Realm.Object {
 
     return newLocationId !== currentLocationId;
   }
+
+  applyLocation(database, newLocation) {
+    if (!this.shouldApplyLocation(newLocation)) return null;
+
+    this.location = newLocation;
+
+    return createRecord(database, 'LocationMovement', this, newLocation);
+  }
+
+  applyVvmStatus(database, newVvmStatus) {
+    return this.shouldApplyVvmStatus(newVvmStatus)
+      ? createRecord(database, 'VaccineVialMonitorStatusLog', this, newVvmStatus)
+      : null;
```

Comment:
same as above. If the vvm status is not different than what it currently is, just do nothing